client/landing/stepper/hooks/use-podcast-title.ts

```diff
@@ -0,0 +1,35 @@
+import apiFetch from '@wordpress/api-fetch';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { useEffect } from 'react';
+import { ONBOARD_STORE } from '../stores';
+import { useAnchorFmParams } from './use-anchor-fm-params';
+
+export default function usePodcastTitle(): string | null {
+	const { siteTitle } = useSelect( ( select ) => select( ONBOARD_STORE ).getState() );
+	const { setSiteTitle } = useDispatch( ONBOARD_STORE );
+	const { anchorFmPodcastId } = useAnchorFmParams();
+	interface PodcastDetails {
+		title?: string;
+	}
+
+	useEffect( () => {
+		if ( ! anchorFmPodcastId ) {
+			return;
+		}
+		// Fetch podcast title from /podcast-details endpoint
+		apiFetch< PodcastDetails >( {
+			path: `https://public-api.wordpress.com/wpcom/v2/podcast-details?url=https://anchor.fm/s/${ encodeURIComponent(
+				anchorFmPodcastId
+			) }/podcast/rss&_fields=title`,
```

Comment:
It seems it doesn't work unless the user enabled the rss feed on the anchor side?

Reply:
It's possible, unfortunately Anchor doesn't have a public API so we rely on their RSS feed to grab episodes AFAIK.

Reply:
(I should note, most of this was pulled over from Gutenboarding, I've just moved individual hooks into separate files for clarity. We're hooking up existing API calls to the new UI to make this functional, so the back end/API piece hasn't changed!)

Reply:
I suspect that on Anchor's side when the link is generated, they have access to the podcast ID regardless of whether the user has activated RSS or not. For our testing purposes, however, the only place we can get the ID is from that URL.